lightning/src/ln/channel.rs

```diff
@@ -739,6 +739,19 @@ pub(super) struct ChannelContext<SP: Deref> where SP::Target: SignerProvider {
 	#[cfg(not(test))]
 	closing_fee_limits: Option<(u64, u64)>,
 
+	/// If we remove an HTLC (or fee update), commit, and receive our counterparty's
+	/// `revoke_and_ack`, we remove all knowledge of said HTLC (or fee update). However, the latest
+	/// local commitment transaction that we can broadcast still contains the HTLC (or old fee),
+	/// thus we are not eligible for initiating the `closing_signed` negotiation if we're shutting
+	/// down and are generally expecting a counterparty `commitment_signed` immediately.
+	///
+	/// To ensure we don't send a `closing_signed` too early, we track this state here, waiting
+	/// until we see a `commitment_signed` before doing so.
+	///
+	/// We don't bother to persist this - we anticipate this state won't last longer than a few
```

Comment:
If your peer goes offline, or withhold `revoke_and_ack`, the state might persist more than few milliseconds, so what should LDK user does, like unilaterally force-close, maybe document `maybe_propose_closing_signed`, I think ?

Reply:
I would favor to add documentation, at the very least that type of silent withhold of your counterparty sucks :(